include/deal.II/grid/filtered_iterator.h

```diff
@@ -960,10 +960,8 @@ FilteredIterator<BaseIterator> &
 FilteredIterator<BaseIterator>::
 operator = (const FilteredIterator &fi)
 {
-  Assert ((fi.state() != IteratorState::valid) || (*predicate)(fi),
-          ExcInvalidElement(fi));
```

Comment:
The same checks are performed in `operator=(const BaseIterator &)` which is now called.